### PR TITLE
Xyz cv api user role referencing

### PR DESCRIPTION
**Pull request by aolundin**
Can now delete user by id and get user by id.
Can update user's role by id using a PUT request.

### CODE REVIEW

**Comment by letterix** [comment]
Hmm, why are we using specific functions for updating a role? Since it is just an field on the user, will we do the same for any field?

If not, what would be the difference between this set of functions, and the functions needed for other fields?

**Comment by aolundin** [comment]
Yup, I forgot to rename the functions. only the function
exports.changeRoleForUser = function(user) in userDao should be specific for roles.

**Comment by letterix** [comment]
And why does it need to be specific? :)

**Comment by letterix** [comment]
Remember, logic should go into the controllers (the valodation etc). The dao should only be our db layer

**Comment by aolundin** [comment]
yea. I should move some logic to controllers and then nothing has to be specific and we'll just have something like a switch statement

**Comment by letterix** [comment]
My thought atleast, you should use the existing validation method for users; we will expand it to use rights aswell

**Comment by Tobbe** [comment]
When you have the original user, and an object with new properties (like a new role for example) something like this might come in handy:

```
function extend(obj, props) {
    for (var prop in props) {
        if (props.hasOwnProperty(prop)) {
            obj[prop] = props[prop];
        }
    }
}

extend(userFromDB, updatedUserProperties);
```

Perhaps there should be some kind of check as well so that `updatedUserProperties` doesn't contain any new properties not available on `userFromDB`

**Comment by aolundin** [comment]
logic moved to controllers

File: app/controllers/user.controller.js [hunk 1]
```diff
@@ -16,11 +16,75 @@ function validateUser(user) {
...
 exports.getUserTemplate = function(name, email) {
     return {
         email: email,
-        name: name
+        name: name,
+        role: 'user'
+    };
+};
+
+function updateUserObject(body) {
```
Comment (letterix) [inline app/controllers/user.controller.js]: Should be named updateUser to follow current naming convention? see createUser

File: app/dao/user.dao.js [hunk 2]
```diff
@@ -19,6 +19,29 @@ exports.createNewUser = function(user) {
         .then(responseHandler.parsePost);
 };
 
+exports.changeFieldForUser = function(user) {
```
Comment (letterix) [inline app/dao/user.dao.js]: should be named updateUser following the current naming convention?

File: app/utils/response.handler.js [hunk 3]
```diff
@@ -64,8 +73,13 @@ exports.parsePost = function(response) {
 
 exports.parsePut = function(response) {
     return q.promise(function(resolve, reject) {
+        if (!response) {
+            return errorHandler.getHttpError(406)
+                .then(reject);
+        }
+
         if (response[0].statusCode === 204) {
-            return resolve(response[1]);
```
Comment (letterix) [inline app/utils/response.handler.js:68]: Hmm why?

File: app/controllers/user.controller.js [hunk 4]
```diff
@@ -16,11 +16,75 @@ function validateUser(user) {
...
+    };
+}
+
+function getUserByIdFunction(id) {
+    return function() {
+        return exports.getUserById(id);
     };
+}
+
+exports.changeFieldForUser = function(request) {
```
Comment (letterix) [inline app/controllers/user.controller.js]: Should be named updateUser to follow current naming convention? see createUser

File: app/dao/user.dao.js [hunk 5]
```diff
@@ -19,6 +19,29 @@ exports.createNewUser = function(user) {
...
+    var options = {
+        uri: url + '/' + user._id,
+        method: 'PUT',
+        json: user
+    };
+
+    return q.nfcall(request, options)
+        .then(responseHandler.parsePut);
```
Comment (letterix) [inline app/dao/user.dao.js:30]: should call responseHandler.parseResponse first

**Comment by aolundin** [comment]
enhanced validation and renamed functions

**Pull request merged**